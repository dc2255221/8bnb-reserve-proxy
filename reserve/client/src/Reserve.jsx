/* eslint-disable linebreak-style */
/* eslint-disable import/extensions */
import React from 'react';
import PriceHeader from './PriceHeader.jsx';
import BookForm from './BookForm.jsx';
import styles from './Reserve.module.css';

class Reserve extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      spaceId: this.props.spaceId,
      nightlyRate: 200,
    };
  }

  // componentDidMount() {
  // }

  render() {
    return (
      <div className={styles.wrapper}>
        <PriceHeader nightlyRate={this.state.nightlyRate} />
        <BookForm />
      </div>
    );
  }
}

export default Reserve;
