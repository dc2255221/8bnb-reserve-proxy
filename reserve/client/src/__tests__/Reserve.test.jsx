import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Reserve from '../Reserve';

describe('<Reserve />', () => {
  // describe a Reserve component with its state
  const space = {
    spaceId: 12,
    nightlyRate: 125,
    cleaningFee: 22.99,
    serviceFee: 40,
    taxRate: 4,
    maxAdultGuests: 2,
    minStayNights: 2,
  };

  // const reservations = [
  //   {
  //     checkinDate: '2020-07-01',
  //     checkoutDate: '2020-07-04',
  //     spaceId: 12,
  //   },
  // ];

  it('matches the snapshot', () => {
    const tree = renderer.create(<Reserve />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains spaceId in state', () => {
    const wrapper = shallow(<Reserve spaceId={space.spaceId} />);
    expect(wrapper.state().spaceId).toBe(12);
  });
});

// write state tests later
// describe('<Reserve /> functions', () => {
//   it('successfully get pricing info', () => {
//     const wrapper = shallow(<Reserve spaceId="15" />);
//     expect(wrapper.state().nightlyRate).toBeLessThanOrEqual(200);
//   });
// });
