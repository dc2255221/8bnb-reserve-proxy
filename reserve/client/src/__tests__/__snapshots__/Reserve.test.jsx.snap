// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reserve /> matches the snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="container"
  >
    <div
      className="price-info"
    >
      <span
        className="price"
      >
        $
        200
      </span>
       
      <span
        className="type"
      >
        per night
      </span>
    </div>
    <div
      className="review-info"
    >
      <span
        className="star"
      />
      <span
        className="rating"
      >
         
        4.75
         
      </span>
      <span
        className="reviews"
      >
        (
        200
         reviews)
      </span>
    </div>
  </div>
  <form
    className="form"
  >
    <div
      className="fieldSet"
    >
      <label>
        <span>
          Dates
        </span>
      </label>
      <div
        className="container"
      >
        <input
          placeholder="Check-in"
          value=""
        />
        <div>
          <svg
            aria-hidden="true"
            className="arrow"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="m0 12.5a.5.5 0 0 0 .5.5h21.79l-6.15 6.15a.5.5 0 1 0 .71.71l7-7v-.01a.5.5 0 0 0 .14-.35.5.5 0 0 0 -.14-.35v-.01l-7-7a .5.5 0 0 0 -.71.71l6.15 6.15h-21.79a.5.5 0 0 0 -.5.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <input
          placeholder="Checkout"
        />
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="box"
        >
          <div
            className="btn"
            name="prev"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <svg
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M336 275L126 485h806c13 0 23 10 23 23s-10 23-23 23H126l210 210c11 11 11 21 0 32-5 5-10 7-16 7s-11-2-16-7L55 524c-11-11-11-21 0-32l249-249c21-22 53 10 32 32z"
              />
            </svg>
          </div>
          <div
            className="text"
          >
            February 2020
          </div>
          <div
            className="btn"
            name="next"
            onClick={[Function]}
            role="button"
            tabIndex="0"
          >
            <svg
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
        </div>
        <div>
          <div
            className="header"
          >
            <ul>
              <li>
                <small>
                  Su
                </small>
              </li>
              <li>
                <small>
                  Mo
                </small>
              </li>
              <li>
                <small>
                  Tu
                </small>
              </li>
              <li>
                <small>
                  We
                </small>
              </li>
              <li>
                <small>
                  Th
                </small>
              </li>
              <li>
                <small>
                  Fr
                </small>
              </li>
              <li>
                <small>
                  Sa
                </small>
              </li>
            </ul>
          </div>
          <table
            className="calendar"
          >
            <tbody>
              <tr
                className="row"
              >
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  26
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  27
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  28
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  29
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  30
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  31
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  1
                </td>
              </tr>
              <tr
                className="row"
              >
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  2
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  3
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  4
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  5
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  6
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  7
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  8
                </td>
              </tr>
              <tr
                className="row"
              >
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  9
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  10
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  11
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  12
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  13
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  14
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  15
                </td>
              </tr>
              <tr
                className="row"
              >
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  16
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  17
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  18
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  19
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  20
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  21
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  22
                </td>
              </tr>
              <tr
                className="row"
              >
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  23
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  24
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  25
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  26
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  27
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  28
                </td>
                <td
                  aria-disabled="true"
                  className="cell"
                  role="button"
                  tabIndex="-1"
                >
                  29
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          className="btnText"
        >
          <button
            className="btn"
            type="button"
          >
            Clear dates
          </button>
        </div>
      </div>
    </div>
    <div
      className="fieldSet"
    >
      <label>
        <span>
          Guests
        </span>
      </label>
      <div>
        <div
          className="container"
        >
          <button
            className="btn"
            type="button"
          >
            GuestPicker dropdown
          </button>
        </div>
        <div
          className="dropdown"
        >
          <div
            className="wrapper"
          >
            <div>
              <div>
                Adults
              </div>
              <div />
            </div>
            <div
              className="item"
            >
              <button
                className="btn"
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="item"
            >
              0
            </div>
            <div
              className="item"
            >
              <button
                className="btn"
                type="button"
              >
                +
              </button>
            </div>
          </div>
          <div
            className="wrapper"
          >
            <div>
              <div>
                Children
              </div>
              <div>
                Ages 2–12
              </div>
            </div>
            <div
              className="item"
            >
              <button
                className="btn"
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="item"
            >
              0
            </div>
            <div
              className="item"
            >
              <button
                className="btn"
                type="button"
              >
                +
              </button>
            </div>
          </div>
          <div
            className="wrapper"
          >
            <div>
              <div>
                Infants
              </div>
              <div>
                Under 2
              </div>
            </div>
            <div
              className="item"
            >
              <button
                className="btn"
                type="button"
              >
                -
              </button>
            </div>
            <div
              className="item"
            >
              0
            </div>
            <div
              className="item"
            >
              <button
                className="btn"
                type="button"
              >
                +
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <input
      type="hidden"
      value={1}
    />
    <input
      type="hidden"
      value={0}
    />
    <input
      type="hidden"
      value={0}
    />
    <div>
      <button
        className="btn"
        type="submit"
      >
        Reserve
      </button>
      <div>
        You won't be charged yet
      </div>
    </div>
  </form>
</div>
`;
